### PR TITLE
MP-5703: Add persistence acl documentation

### PR DESCRIPTION
**Pull request by ievgen-malykhin**
## PR Description

TBD

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/execution-flow.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+If the user does not have enough permissions during query execution, the system will change the query so that the result of executing will be an empty collection.
+![Query execution flow](https://confluence-connect.gliffy.net/embed/image/c84bb011-1c7c-45e7-84b3-f98b2fee8e08.png?utm_medium=live&utm_source=custom)
+
+When processing a query, Persistence ACL will consider the rules according to the following criteria:
+- The entity to which the rule applied  match the entity of the query.
+- The operation to which the rule grants permissions correspond to the operation of the query.
+
+If the remaining rules have different scopes, only those rules that apply to the highest priority scopes will be applied.
+
+Scope priority
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/execution-flow.md:19]: please mention that this is configurable

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/persistence-acl-feature-walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+Run the command to apply attached behavior
+```bash
+console propel:install
+```
+
+## Module dependency graph
+![Module dependency graph](https://confluence-connect.gliffy.net/embed/image/b15ac7bf-e35f-4298-90da-b7d0c8227be9.png?utm_medium=live&utm_source=custom)
+
+## Domain model
+![Domain model](https://confluence-connect.gliffy.net/embed/image/4fe4c0ba-1192-4aca-97f8-d996dfccc583.png?utm_medium=live&utm_source=custom)
```
[the remark this answers is not in the document]
Comment (ievgen-malykhin) [reply docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/persistence-acl-feature-walkthrough.md:54]: We consider to keep pivot tables. From one side they make the diagram more complicated, but from the other they are significant for persistence ACL feature.

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/rules-and-scopes.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
+---
```
[the remark this answers is not in the document]
Comment (ievgen-malykhin) [reply docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/rules-and-scopes.md:1]: splitted to global, inherited, segment and composite entity

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/rules-and-scopes.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+Query after Persistence ACL
+```sql
+SELECT `spy_merchant_profile`.* 
+FROM `spy_merchant_profile`
+  INNER JOIN `spy_merchant` ON (`spy_merchant_profile`.`fk_merchant` = `spy_merchant`.`id_merchant`)
+  INNER JOIN `spy_acl_entity_segment_merchant`
+    ON (`spy_merchant`.`id_merchant` = `spy_acl_entity_segment_merchant`.`fk_merchant` 
+      AND `spy_acl_entity_segment_merchant`.`fk_acl_entity_segment` IN (18)); 
+```
```
[the remark this answers is not in the document]
Comment (ievgen-malykhin) [reply docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/rules-and-scopes.md:249]: @ihorsamusenko what do you think if I mention "this is just enough to have R rule for a parent" in Inherited scope page? Looks like here is nothing to compare

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/persistence-acl-feature-walkthrough.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
+---
+title: Persistence ACL feature walkthrough
+last_updated: Sep 14, 2021
+template: concept-topic-template
+---
+
+## Overview
+Persistence ACL feature allows to manage authorisation at the database entity level, and even a set of entities (segments). 
+This feature supports a flexible system of inheritance of rights, which simplifies the configuration of access in the system. 
+We will talk about it, as well as about other capabilities of the module below.
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/persistence-acl-feature-walkthrough.md:10]:
```suggestion
We will talk about it, as well as about other capabilities of the feature below.
```

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/persistence-acl-feature-walkthrough.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+As the name suggests, the Persistence ACL runs in the Persistence layer.
+
+## Limitations
+The module based on Propel ORM (namely Propel Behavior and Propel Hooks). If you do not use PropelOrm to interact with data in your system, this module will not work.
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/persistence-acl-feature-walkthrough.md:14]:
```suggestion
The module based on Propel ORM (namely Propel Behavior and Propel Hooks). If you do not use PropelOrm to interact with data in your system, this feature will not work for you.
```

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/persistence-acl-feature-walkthrough.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+- preSelectQuery
+- preUpdateQuery
+- preDeleteQuery
+
+When a query to the database is issued, the query will be intercepted and modified with additional joins to narrow down the result of the query to only the records the are available to the current user.
+If the user tries to perform a restricted action on an Active Record model (such an update, delete or create),
+an `\Spryker\Zed\AclEntity\Persistence\Exception\OperationNotAuthorizedException will be thrown.`
+
+## Learn more
+- Configuration <!---LINK-->
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/persistence-acl-feature-walkthrough.md:53]: why do you leave as ? this pages are here in the PR

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+last_updated: Sep 14, 2021
+template: concept-topic-template
+---
+
+Persistence ACL feature based functionality on Propel behavior. It's possible to enable the feature in 2 ways:
+- Connect the feature to one or several tables in the database
+- Connect the feature to all tables in the database
+
+## Connect Persistence ACL feature to one or several tables
+```xml
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md:12]: Please mention smth like: In the following code snippet only SpyMerchant entity is configured to be handled by ACL.

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+          xsi:schemaLocation="spryker:schema-01 https://static.spryker.com/schema-01.xsd" 
+          namespace="Orm\Zed\Merchant\Persistence"
+          package="src.Orm.Zed.Merchant.Persistence">
+    <table name="spy_merchant">
+      <behavior name="\Spryker\Zed\AclEntity\Persistence\Propel\Behavior\AclEntityBehavior"/>
+    </table>
+</database>
+```
+
+## Connect Persistence ACL feature to all tables
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md:26]: Please mention smth like: in the following example _all_ the entities on the system will be handled by ACL. When you do such a configuration dont forget to use AllowList configuration (here should go a link) to exclude entities that are needed by ACL to function properly. Provide the list of entities that are needed (SpyRole, SpyEntiyRule, and all the other that are required).

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+
+```
+## Feature configuration
+![Configuration entity relation diagram](https://confluence-connect.gliffy.net/embed/image/f2309504-8638-419d-abf9-783bc45c8792.png?utm_medium=live&utm_source=custom)
+The configuration, unlike the rule, is common to the system.
+The main configuration object of the module is `\Generated\Shared\Transfer\AclEntityMetadataConfigTransfer`.
+It's available for extension through the plugin system.
+All you need is create a plugin and implement `\Spryker\Zed\AclEntityExtension\Dependency\Plugin\AclEntityMetadataConfigExpanderPluginInterface`.
+
+`\Generated\Shared\Transfer\AclEntityMetadataConfigTransfer`
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md:46]: can we have it as h3 or h4 maybe?

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+|-----|-----|-----|
+| collection | `\Generated\Shared\Transfer\AclEntityMetadataTransfer[]` | The set of configurations for the models. |
+
+`\Generated\Shared\Transfer\AclEntityMetadataTransfer`
+
+| property | type | description |
+|-----|-----|-----|
+| parent | `\Generated\Shared\Transfer\AclEntityParentMetadataTransfer` | This property is used to configure inheritance. Required for entity which has rules with [Inherited scope](rules-and-scopes/inherited-scope.html), or for [Composite entity](rules-and-scopes/composite-entity.html) (See [Inherited scope vs Composite entity](rules-and-scopes/composite-entity.html#inherited-scope-vs-composite-entity)). |
+| entityName | string | Fully qualified class name of configured entity (Propel Entity). |
+| hasSegmentTable | bool | Sets if configured entity supports segmentation (see [Segment scope documentation](rules-and-scopes/segment-scope.htmlSets the default binary access mask)). |
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md:65]:
```suggestion
| hasSegmentTable | bool | Sets if configured entity supports segmentation (see [Segment scope documentation](rules-and-scopes/segment-scope.html) Sets the default binary access mask)). |
```

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+| isSubentity | bool | Indicates whether the configured entity is part of a composite object (see [Composite entity](rules-and-scopes/composite-entity.html) documentation). |
+
+`\Generated\Shared\Transfer\AclEntityParentMetadataTransfer`
+
+| property | type | description |
+|-----|-----|-----|
+| connection | `\Generated\Shared\Transfer\AclEntityParentConnectionMetadata` | This property is used to set up the relationship between the current class and the parent. |
+| entityName | string | Fully qualified class name of parent entity |
+
+Sometimes the links between the child and parent tables established not through foreign keys, but using so-called "reference columns".
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md:76]: I think this part should go under `\Generated\Shared\Transfer\AclEntityParentConnectionMetadataTransfer`?

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/execution-flow.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+title: Execution flow
+last_updated: Sep 14, 2021
+template: concept-topic-template
+---
+## Query processing flow
+Query processing flow
+
+The process of selecting and applying a rule for a query and when performing a model action are similar, but still they have some difference.
+If the user does not have enough permissions during query execution, the system will change the query so that the result of executing will be an empty collection.
+![Query execution flow](https://confluence-connect.gliffy.net/embed/image/c84bb011-1c7c-45e7-84b3-f98b2fee8e08.png?utm_medium=live&utm_source=custom)
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/execution-flow.md:11]: wrong diagram?

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/rules-and-scopes/rules-and-scopes.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+Each CRUD operation has its own binary mask.
+
+| Operation | Binary mask | Integer representation |
+|-----|-----|-----|
+| Read |`0b1` | 1 |
+| Create |`0b10` | 2  |
+| Update |`0b100` | 4 |
+| Delete |`0b1000` | 8 |
+
+See [Spryker\Shared\AclEntity\AclEntityConstants](https://github.com/spryker/acl-entity/blob/master/src/Spryker/Shared/AclEntity/AclEntityConstants.php) for details.
```
Comment (ihorsamusenko) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/rules-and-scopes/rules-and-scopes.md:67]: Please add here a section with links to scopes docs (global composite segment and inherited)

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/rules-and-scopes.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ievgen-malykhin) [reply docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/rules-and-scopes.md]: @ihorsamusenko review the table after the fixes please.

File: docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md (hunk 10 above)
Comment (ievgen-malykhin) [reply docs/marketplace/dev/feature-walkthroughs/202108.0/persistence-acl-feature-walkthrough/configuration.md:46]: ### was made. looks not so good in navigation bar due the long namespaces.

**Review by gibmirmilch — APPROVED** [review]

**Pull request merged**